src/Modules/UngradedModules/FreeModElem.jl

```diff
@@ -254,28 +254,20 @@ function Base.deepcopy_internal(a::AbstractFreeModElem, dict::IdDict)
 end
 
 # scalar multiplication with polynomials, integers
-function *(a::MPolyDecRingElem, b::AbstractFreeModElem)
-  @req parent(a) === base_ring(parent(b)) "elements not compatible"
-  return parent(b)(a*coordinates(b))
-end
+*(a::Any, b::AbstractFreeModElem) = parent(b)(base_ring(parent(b))(a)*coordinates(b))
 
-function *(a::MPolyRingElem, b::AbstractFreeModElem) 
-  if parent(a) !== base_ring(parent(b))
-    return base_ring(parent(b))(a)*b # this will throw if conversion is not possible
-  end
-  return parent(b)(a*coordinates(b))
+function *(a::T, b::AbstractFreeModElem{T}) where {T <: NCRingElem}
+  parent(a) === base_ring(parent(b)) && return parent(b)(a*coordinates(b))
+  return parent(b)(base_ring(parent(b))(a)*coordinates(b))
 end
 
-function *(a::RingElem, b::AbstractFreeModElem) 
-  if parent(a) !== base_ring(parent(b))
-    return base_ring(parent(b))(a)*b # this will throw if conversion is not possible
-  end
-  return parent(b)(a*coordinates(b))
+function *(b::AbstractFreeModElem{T}, a::T) where {T <: NCRingElem}
+  error("all modules are left-modules by default")
 end
 
-*(a::Int, b::AbstractFreeModElem) = parent(b)(a*coordinates(b))
-*(a::Integer, b::AbstractFreeModElem) = parent(b)(base_ring(parent(b))(a)*coordinates(b))
-*(a::QQFieldElem, b::AbstractFreeModElem) = parent(b)(base_ring(parent(b))(a)*coordinates(b))
+function *(b::AbstractFreeModElem{T}, a::Any) where {T <: RingElem}
+  error("scalar multiplication from the right is not defined; use left-multiplication instead")
+end
```

Comment:
```suggestion
function *(b::AbstractFreeModElem{T}, a::Any) where {T <: RingElem}
  error("scalar multiplication from the right is not yet supported")
end

# Methods to determine whether a module is a left-, right-, or bi-module. 
# We plan to have flags set for this. But for the moment the generic code only supports left-multiplication, 
# so we can decide this from the type alone. How we do it in the long run is not yet decided, but in either case
# we want to use these functions to decide as they are already there for ideals. 
is_left(M::ModuleFP) = is_left_module(typeof(M))
is_left(::Type{T}) where {RET<:RingElem, T<:ModuleFP{T}} = true
is_left(::Type{T}) where {RET<:NCRingElem, T<:ModuleFP{T}} = true # Left multiplication is generically supported

is_right(M::ModuleFP) = is_left_module(typeof(M))
is_right(::Type{T}) where {RET<:RingElem, T<:ModuleFP{T}} = true
is_right(::Type{T}) where {RET<:NCRingElem, T<:ModuleFP{T}} = false # Right multiplication is not supported by the generic code at the moment, but we plan to do so eventually. 

is_two_sided(M::ModuleFP) = is_left_module(typeof(M))
is_two_sided(::Type{T}) where {RET<:RingElem, T<:ModuleFP{T}} = true
is_two_sided(::Type{T}) where {RET<:NCRingElem, T<:ModuleFP{T}} = false # see above

```

Reply:
In L280 and L284 I think you forgot to update the rhs when copying it there